graphs/util/gpml_util_interface.py

```diff
@@ -0,0 +1,296 @@
+        # temp_node['data']['background_opacity'] = 0
+        # increase count, i.e. to keep unique ids for shapes, nodes, labels.
+        count = count + 1
+    return G,   count
+
+
+def parse_interactions(G,   count, interactions):
+    # Create a list of (x, y) coordinates of an edge
+    # every element of the list can be either of length of 2 or 3
+    # as curved edges are represented by 3 coordinates.
+    edge_details = []
+    for edge in interactions:
+        for details in edge:
+            temp_edge = []
+            for more in details:
+                if 'X' in more.attrib and 'Y' in more.attrib:
+                    temp_edge.append((more.attrib['X'], more.attrib['Y']))
+            edge_details.append(temp_edge)
+
+    # the above way of adding attributes will append after every empty list
+    # something like [['something'], [], ['more'], [], ['something']]
+    # that's why we use the magic of list slicing [::2]
+    # create temp edge ids
+    temp_edge_ids = []
+    for edge in edge_details[::2]:
+        edge_id = []
+        for nodes in edge:
+            temp_node = {'data': {}}
+            interface.add_node(G, count)
+            edge_id.append(count)
+            interface.add_node_x_coordinate(G, count, nodes[0])
+            interface.add_node_y_coordinate(G, count, nodes[1])
+            interface.add_node_color(G, count, 'white')
+            interface.add_node_background_opacity(G, count, 0)
+            count = count + 1
+        temp_edge_ids.append(edge_id)
+
+    # create edges
+    # TODO: extract more information from edges, i.e arrow shapes
+    # right now by default we are using triangle as arrow shape.
+    for edge in temp_edge_ids:
+        temp_edge = {'data': {}}
+        if len(edge) == 2:
+            interface.add_edge(G, edge[0], edge[1])
+            interface.add_edge_target_arrow_shape(G, edge[0], edge[1], 'triangle')
+        if len(edge) == 3:
+            interface.add_edge(G, edge[0], edge[1])
+            interface.add_edge_target_arrow_shape(G, edge[0], edge[1], 'triangle')
+
+            temp_edge = {'data': {}}
+            interface.add_edge(G, edge[1], edge[2])
+
+    # This is the where magic happens, creating the layout
+    # for the GPML file.
+    return G, count
+
+
+def set_gpml_layout(G):
+    default_layout = []
+    for node in G.nodes_iter(data=True):
+        temp = dict()
+        temp['id'] = str(node[0])
+        temp['x'] = float(node[1]["x"])
+        temp['y'] = float(node[1]["y"])
+        default_layout.append(temp)
+
+    # Well, we need it in JSON format. So ..
+    default_layout = json.dumps(default_layout)
+    return default_layout
+
+
+def parse_groups(G, groups):
+    for group in groups:
+        temp_node = {'data': {}}
+        interface.add_node(G, group.attrib['GroupId'])
+        interface.add_node_color(G,group.attrib['GroupId'], 'white')
+        interface.add_node_border_color(G, group.attrib['GroupId'], 'black')
```

Comment:
Add this comment: We explicitly make the border of a group to be black so that we can see the group in GraphSpace.